### DISCUSSION TITLE
Pages not syncing behind reverse proxy (Apache)

### DISCUSSION THREAD

**Discussion by BrendanTCC** [docmost/docmost#155] [Help / Questions] [answered]
Hey gang,

Right off the bat it seems as if the issue is websockets (based on the discussion in #133, #110, and #104) however for some reason I thought it to be a grand idea to use Apache as my reverse proxy.

Because I don't usually work with websockets, let alone within Apache, I'll openly admit I don't know what the hell I'm doing.

In the spirit of getting this documented, thought I'd log this in the hope that someone who's played with reverse proxying WebSockets via Apache before stumbles across this.

**Comment by BrendanTCC** [comment, marked as answer]
...of course it's documented on the Grafana forums of all places - Just in case someone stumbles across this, as long as you make sure this:

```
    RewriteEngine on
    RewriteCond %{HTTP:Upgrade} websocket [NC]
    RewriteCond %{HTTP:Connection} upgrade [NC]
    RewriteRule ^/?(.*) "ws://127.0.0.1:3000/$1" [P,L]
```

Is in your config file (along with the usual reverse proxy stuff), you should be fine.

Will leave this open for now just in case this is better converted into a Discussion, unless there's things that might be worth adding in (though I realize this could be done even with the issue closed... probably)

**Comment by triplenick** [comment]
**NGINX** solution, configuration file:

```
server {
    listen 80;
    server_name yourdomain.com;  # Replace with your domain

    location / {
        proxy_pass http://localhost:[YOUR_DOCMOST_PORT_HERE];
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        # WebSocket specific headers
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
}
```

**Comment by jun3010me** [comment]
This worked in my environment.

https://github.com/jun3010me/Apache2_ReverseProxy_config_for_Docmost/blob/main/yourdomain.conf
